src/CodeViewer/ViewerTest.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <QApplication>
+#include <QFont>
+#include <QFontDatabase>
+#include <QFontMetrics>
+#include <array>
+#include <limits>
+
+#include "CodeViewer/Viewer.h"
+#include "gtest/gtest.h"
+
+namespace orbit_code_viewer {
+
+TEST(Viewer, DecimalDigitsFromNumber) {
+  EXPECT_EQ(DecimalDigitsFromNumber(0), 1);
+  EXPECT_EQ(DecimalDigitsFromNumber(1), 1);
+  EXPECT_EQ(DecimalDigitsFromNumber(9), 1);
+  EXPECT_EQ(DecimalDigitsFromNumber(10), 2);
+  EXPECT_EQ(DecimalDigitsFromNumber(123), 3);
+  EXPECT_EQ(DecimalDigitsFromNumber(123000), 6);
+
+  EXPECT_EQ(DecimalDigitsFromNumber(-1), 1);
+  EXPECT_EQ(DecimalDigitsFromNumber(-10), 2);
+
+  // This mainly checks whether the function crashes with very large numbers.
+  // Correctness is a minor concern.
+  EXPECT_GE(DecimalDigitsFromNumber(std::numeric_limits<int>::max()), 10);
+  EXPECT_GE(DecimalDigitsFromNumber(std::numeric_limits<int>::min() / 2), 10);
+}
+
+static int argc = 0;
+
+TEST(Viewer, DetermineLineNumberWidthInPixels) {
+  QApplication app{argc, nullptr};
+
+  const QFont font = QFontDatabase::systemFont(QFontDatabase::FixedFont);
+  const QFontMetrics font_metrics{font};
+
+  const std::array max_line_numbers = {20, 99, 333, 999, 2000};
+
+  for (const int max_line_number : max_line_numbers) {
+    const int number_of_pixels_needed =
+        DetermineLineNumberWidthInPixels(font_metrics, max_line_number);
+
+    // All the line numbers between 1 and <max_line_number> need to fit
+    // in the width calculated by DetermineLineNumberWidthInPixels.
+    for (int line_number = 1; line_number <= max_line_number; ++line_number) {
+      const auto line_number_label = QString::number(line_number);
+      EXPECT_LE(font_metrics.horizontalAdvance(line_number_label), number_of_pixels_needed);
+    }
```

Comment:
Very interesting, this test fails for me if I execute it by calling `build_default/bin/CodeViewerTests`. It does not fail when i `ctest` in the build folder

Reply:
This behaviour is only on Linux. On Windows the test is always successful, independent from how its started.

Reply:
Great that it fails. This test is testing for cut-off digits, so it proves for problem from above.

Reply:
I changed the code. Could you please test again? I tried to reproduce this on my linux machine and it was fine there. So it's hard for me to figure out if it's working.

Reply:
All unit tests successful. Independent from how I started them. 
Looking at the code, it also makes sense. Simple but effective change!